refactor(queries): Split schema.ts for maintainability

Twixes opened the pull request:
## Problem

`schema.ts` has been growing without bounds as a monolith, like `types.ts` has for even longer. This has been necessary, as `ts-json-schema-generator` didn't support `export * from ...`, so we couldn't split the file.
Until `ts-json-schema-generator` 2.2.0, which added support for wildcard export. Finally, the technology to split the schema into sensible chunks.

## Changes

Proposing we split `schema.ts` into:
`schema/schema-assistant-messages.ts` – AI assistant messages
`schema/schema-assistant-queries.ts` – AI assistant-tuned queries
`schema/schema-general.ts` – basically all the regular queries and their building blocks
`schema/type-utils.ts` - common util types that don't need to be rendered into Python as their own types
`schema/index.ts` - `export *`s of the above for Python translation, except utils

## How did you test this code?

All existing tests should continue to pass.

Twixes committed:
refactor(queries): Split schema.ts for maintainability

skoob13 commented on posthog/schema.py:
```diff
@@ -1094,13 +897,8 @@ class InsightFilterProperty(StrEnum):
-    STICKINESS_QUERY = "StickinessQuery"
-    LIFECYCLE_QUERY = "LifecycleQuery"
+class InsightNodeKind(RootModel[str]):
+    root: str
```
Enums got lost

Twixes replied:
I dug into this and painfully it's a regression in `ts-json-schema-generator` that went in _just before_ support for `export *` went in: https://github.com/vega/ts-json-schema-generator/pull/1927. Since I've already contributed to `ts-json-schema-generator` before, I took to just putting out a PR with the fix myself: https://github.com/vega/ts-json-schema-generator/pull/2149

Twixes replied:
Hahaha, now `ts-json-schema-generator` has support for `export *`, but we instead find out `kea-typegen` is confused by it (results in duplicated imports, CC @mariusandra in case this already was on your radar)

Twixes replied:
PR here: https://github.com/keajs/kea-typegen/pull/50

thmsobrmlr committed:
use ts-json-schema-generator v2.4.0-next.6

thmsobrmlr commented:
Seems to work with the latests pre-release. Needs to be updated for changes to schema.ts though

Twixes committed:
Merge branch 'master' into split-schema

Twixes committed:
Rebuild schema

Twixes committed:
Upgrade kea-typegen

Twixes committed:
Merge branch 'master' into split-schema

Twixes committed:
Override TypeScript

Latest `ts-json-schema-generator` is on TypeScript 5.6, which no longer has `suppressImplicitAnyIndexErrors`. We're relying on `suppressImplicitAnyIndexErrors` a bit too much right now to rip it out though, so pinning everything to the older TS 4.9.

Twixes committed:
Merge branch 'master' into split-schema

Twixes committed:
Fix some imports

Twixes committed:
Fix more imports

Twixes committed:
Update utils.ts

Twixes committed:
Update chatResponse.mocks.ts
